api/profile/profileRouter.js

```diff
@@ -128,28 +128,23 @@ router.get('/', authRequired, adminRequired, function (req, res, next) {
  *      404:
  *        description: 'Profile not found'
  */
-router.get(
-  '/:id',
-  authRequired,
-
-  async function (req, res, next) {
-    const id = String(req.params.id);
-    const attendance_average = await Profiles.checkAverageAttendance(id);
-    Profiles.findById(id)
-      .then((profile) => {
-        if (profile) {
-          res
-            .status(200)
-            .json({ ...profile, attendance_rate: attendance_average });
-        } else {
-          next({ status: 404, message: 'ProfileNotFound' });
-        }
-      })
-      .catch((err) => {
-        next({ status: 500, message: err.message });
-      });
-  }
-);
+router.get('/:id', authRequired, async function (req, res, next) {
+  const id = String(req.params.id);
+  const attendance_average = await Profiles.checkAverageAttendance(id);
+  Profiles.findById(id)
+    .then((profile) => {
+      if (profile) {
+        res
+          .status(200)
+          .json({ ...profile, attendance_rate: attendance_average });
+      } else {
+        next({ status: 404, message: 'ProfileNotFound' });
```

Comment:
404 mean's there is no address at that location. Meaning if you called `/users/:id` and there was no endpoint for it your server would give a 404. 

Let's take a look at a better error for a missing 'profileById'.